warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/WhindexFieldIngestHelper.java

```diff
@@ -0,0 +1,247 @@
+                    break;
+                case VALUES:
+                    // Split the comma-separated list of values and set them.
+                    whindexConfig.setValues(List.of(entry.getValue().split(",")));
+                    break;
+                default:
+                    // Log a warning for any unexpected property found in the configuration.
+                    String originalProperty = commonPrefix + groupID + "." + property;
+                    log.warn("Unexpected whindex property given:" + originalProperty + "=" + entry.getValue());
+            }
+        }
+
+        // After processing, map each WhindexConfig to its value field.
+        groupingsToConfigs.values().forEach((wc) -> {
+            valueFieldsToWhindexConfigs.put(wc.getValueField(), wc);
+        });
+    }
+
+    /**
+     * Processes the provided event map to generate new whindex fields based on the whindex rules.
+     * <p>
+     * The helper examines each WhindexConfig and checks:
+     * <ul>
+     * <li>That both the value field and source field exist in the input event map.</li>
+     * <li>That at least one of the entries in the event map for the value field matches one of the configured values, either by event field value or indexed
+     * field value.</li>
+     * </ul>
+     * If a match is found, a copy of the source field's NormalizedContentInterface objects is created and added to the whindex fields under the configured
+     * destination field.
+     * </p>
+     *
+     * @param eventMap
+     *            the input event map containing field names and their corresponding normalized content.
+     * @return a multimap of whindex fields generated from the event map.
+     */
+    @Override
+    public Multimap<String,NormalizedContentInterface> getWhindexFields(Multimap<String,NormalizedContentInterface> eventMap) {
+        Multimap<String,NormalizedContentInterface> whindicesInEventMap = HashMultimap.create();
+
+        // Filter whindex rules for which both the value field and source field are present in the event map.
+        List<WhindexConfig> matchingConfigs = valueFieldsToWhindexConfigs.entries().stream()
+                        .filter(entry -> eventMap.containsKey(entry.getValue().getValueField()) && eventMap.containsKey(entry.getValue().getSourceField()))
+                        .map(Map.Entry::getValue).collect(Collectors.toList());
+
+        // Iterate over the matching configurations to generate the whindex fields.
+        for (WhindexConfig currWhindexConfig : matchingConfigs) {
+            // Retrieve all normalized content corresponding to the value field.
+            Collection<NormalizedContentInterface> relatedValueEventContents = eventMap.get(currWhindexConfig.getValueField());
+            // Check if any of these contents match the allowed values (either event field or indexed field).
+            boolean containsAnyMatchingValue = relatedValueEventContents.stream().anyMatch(nci -> {
```

Comment:
This could be very inefficient, lets make one pass through the values as opposed to two when creating the whindex entries.